p256/src/arithmetic/hash2curve.rs

```diff
@@ -0,0 +1,283 @@
+            p_y: hex!("ecb9f0eadc9aeed232dabc53235368c1394c78de05dd96893eefa62b0f4757dc"),
+            u_0: hex!("4ebc95a6e839b1ae3c63b847798e85cb3c12d3817ec6ebc10af6ee51adb29fec"),
+            u_1: hex!("4e21af88e22ea80156aff790750121035b3eefaa96b425a8716e0d20b4e269ee"),
+            q0_x: hex!("d88b989ee9d1295df413d4456c5c850b8b2fb0f5402cc5c4c7e815412e926db8"),
+            q0_y: hex!("bb4a1edeff506cf16def96afff41b16fc74f6dbd55c2210e5b8f011ba32f4f40"),
+            q1_x: hex!("a281e34e628f3a4d2a53fa87ff973537d68ad4fbc28d3be5e8d9f6a2571c5a4b"),
+            q1_y: hex!("f6ed88a7aab56a488100e6f1174fa9810b47db13e86be999644922961206e184"),
+        },
+    ];
+
+    for test_vector in TEST_VECTORS {
+        // in parts
+        let mut u = [FieldElement::default(), FieldElement::default()];
+        hash2field::hash_to_field::<ExpandMsgXmd<Sha256>, FieldElement>(
+            &[test_vector.msg],
+            DST,
+            &mut u,
+        )
+        .unwrap();
+        assert_eq!(u[0].to_bytes().as_slice(), test_vector.u_0);
+        assert_eq!(u[1].to_bytes().as_slice(), test_vector.u_1);
+
+        let q0 = u[0].map_to_curve();
+        let aq0 = q0.to_affine();
+        assert_eq!(aq0.x.to_bytes().as_slice(), test_vector.q0_x);
+        assert_eq!(aq0.y.to_bytes().as_slice(), test_vector.q0_y);
+
+        let q1 = u[1].map_to_curve();
+        let aq1 = q1.to_affine();
+        assert_eq!(aq1.x.to_bytes().as_slice(), test_vector.q1_x);
+        assert_eq!(aq1.y.to_bytes().as_slice(), test_vector.q1_y);
+
+        let p = q0.clear_cofactor() + q1.clear_cofactor();
+        let ap = p.to_affine();
+        assert_eq!(ap.x.to_bytes().as_slice(), test_vector.p_x);
+        assert_eq!(ap.y.to_bytes().as_slice(), test_vector.p_y);
+
+        // complete run
+        let pt =
+            NistP256::hash_from_bytes::<ExpandMsgXmd<Sha256>>(&[test_vector.msg], DST).unwrap();
+        let apt = pt.to_affine();
+        assert_eq!(apt.x.to_bytes().as_slice(), test_vector.p_x);
+        assert_eq!(apt.y.to_bytes().as_slice(), test_vector.p_y);
+    }
+}
+
+#[test]
+fn hash_to_scalar_voprf() {
+    use elliptic_curve::hash2field::ExpandMsgXmd;
+    use hex_literal::hex;
+    use sha2::Sha256;
+
+    struct TestVector {
+        dst: &'static [u8],
+        seed: &'static [u8],
+        sk_sm: &'static [u8],
+    }
+
+    const TEST_VECTORS: &[TestVector] = &[
+        TestVector {
+            dst: b"HashToScalar-VOPRF08-\x00\x00\x03",
+            seed: &hex!("a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3"),
+            sk_sm: &hex!("c15d9e9ab36d495d9d62954db6aafe06d3edabf41600d58f9be0737af2719e97"),
+        },
+        TestVector {
+            dst: b"HashToScalar-VOPRF08-\x01\x00\x03",
+            seed: &hex!("a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3a3"),
+            sk_sm: &hex!("7f62054fcd598b5e023c08ef0f04e05e26867438d5e355e846c9d8788d5c7a12"),
+        },
+    ];
```

Comment:
I copied these test vectors from the VOPRF spec, I don't know any other spec providing test vectors for `hash_to_scalar`. The VOPRF spec also doesn't define any for k256.